src/Symfony/Component/Security/Http/Authenticator/FormLoginAuthenticator.php

```diff
@@ -129,12 +129,20 @@ private function getCredentials(Request $request): array
 
         $credentials['username'] = trim($credentials['username']);
 
+        if ('' === (string) $credentials['username']) {
```

Comment:
I think this type cast is not needed, as it's now after the `trim()` call, which has a return type of `string` according to php.net

```suggestion
        if ('' === $credentials['username']) {
```